### PR TITLE
Add Withdrawal helpers

### PR DESCRIPTION
**Pull request by potuz**
This PR adds Capella helpers for state transition.

### CODE REVIEW

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
Approved. The description is wring though.

**Comment by prestonvanloon** [comment]
Please check that `creds`` is not empty in HasETH1WithdrawalCredential.

```go
return len(cred) > 0 && cred[0] == params.BeaconConfig().ETH1AddressWithdrawalPrefixByte 
```

File: beacon-chain/state/state-native/readonly_validator.go [hunk 1]
```diff
@@ -78,6 +79,28 @@ func (v readOnlyValidator) WithdrawalCredentials() []byte {
 	return creds
 }
 
+// HasETH1WithdrawalCredential returns whether the validator has an ETH1
+// Withdrawal prefix.
+func (v readOnlyValidator) HasETH1WithdrawalCredential() bool {
+	cred := v.WithdrawalCredentials()
+	return cred[0] == params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
```
Comment (prestonvanloon) [inline beacon-chain/state/state-native/readonly_validator.go:86]:
```suggestion
	return len(cred) > 0 && cred[0] == params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
```

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

File: beacon-chain/state/state-native/readonly_validator.go (hunk 1 above)
Comment (potuz) [reply beacon-chain/state/state-native/readonly_validator.go:86]: Why this check? A validator can never have empty withrawal credentials, it's unmarshalled as 32 bytes

Comment (prestonvanloon) [reply beacon-chain/state/state-native/readonly_validator.go:86]: It's a free defensive check and will make fuzzing work better.

Comment (prestonvanloon) [reply beacon-chain/state/state-native/readonly_validator.go:86]: The risk of not having this is that the node will panic in the (presumably) impossible scenario where the credential slice is empty

**Review by terencechain — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**